near/contracts/bridge/src/lib.rs

```diff
@@ -848,18 +982,29 @@ impl FastBridge {
         amount: U128,
         recipient_id: AccountId,
     ) -> U128 {
-        if is_promise_success() {
+        let mut transferred_amount = U128(0);
+
+        if let Some(result) = promise_result_as_success() {
+            transferred_amount = if result.is_empty() {
+                amount
+            } else {
+                near_sdk::serde_json::from_slice::<U128>(&result).unwrap()
+            };
+
             Event::FastBridgeWithdrawEvent {
-                recipient_id,
-                token: token_id,
-                amount,
+                recipient_id: recipient_id.clone(),
```

Comment:
should we add also `sender_id` field?

Reply:
ed05e3d8bd04ba57be1986de6392ca6e8e782b79